src/balance/summary.ts

```diff
@@ -7,18 +8,21 @@ export function assetTransfersToSummary(
   transfers: AssetTransfer[],
 ): FungibleSummary {
   return transfers.reduce((previousValue, currentValue): FungibleSummary => {
-    let tokenSummary = previousValue.get(currentValue.tokenAddress);
+    const { tokenAddress, decimals, symbol } = currentValue;
+    let tokenSummary = previousValue.get(tokenAddress);
     if (typeof tokenSummary === "undefined") {
       tokenSummary = {
-        tokenAddress: currentValue.tokenAddress,
+        tokenAddress,
         amount: "0",
-        decimals: currentValue.decimals,
-        symbol: currentValue.symbol,
+        decimals,
+        symbol,
       };
       previousValue.set(currentValue.tokenAddress, tokenSummary);
     }
-    tokenSummary.amount = (
-      Number(tokenSummary.amount) + Number(currentValue.amount)
+    tokenSummary.amount = formatUnits(
```

Comment:
I think ideally we store the `bigint` value in the `tokenSummary` instead of the decimals.
Then we do not need to `formatUnits` at all anymore.

Reply:
This turned out to be quite a change... 
All contained in https://github.com/bh2smith/multi-asset-transfer/pull/10/commits/30d76dba6bb00cc88d8404caa6ea162fd5792658

Let me know what you think.